<launch>
    <!-- camera parameters -->
    <arg name="camera_name" default="left_camera" />
    <arg name="device_name" default="/dev/video0" />
    <arg name="frame_id" default="$(arg camera_name)" />
    <arg name="width" default="744" />
    <arg name="height" default="480" />
    <arg name="fps" default="30" />
    <arg name="image_format" default="bayer_gbrg8" />
    <arg name="camera_info_url" default="file://$(find lirs_video_stream)/calib/$(arg camera_name).yaml" />

    <!-- image view -->
    <arg name="show" default="true" />
    <arg name="show_topic" default="image_raw"/>

    <!-- image proc -->
    <arg name="proc" default="true" />


    <group ns="$(arg camera_name)">

        <node pkg="lirs_video_stream" type="video_stream" name="$(arg camera_name)_video_stream" output="screen">
            <param name="camera_name" type="string" value="$(arg camera_name)" />
            <param name="device_name" type="string" value="$(arg device_name)" />
            <param name="frame_id" type="string" value="$(arg frame_id)" />
            <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
            <param name="width" type="int" value="$(arg width)" />
            <param name="height" type="int" value="$(arg height)" />
            <param name="fps" type="int" value="$(arg fps)" />
            <param name="image_format" type="string" value="$(arg image_format)" />
            <remap from="image" to="image_raw" />
        </node>

        <node if="$(arg proc)" name="image_proc" pkg="image_proc" type="image_proc">
            <env name="ROS_NAMESPACE" value="$(arg camera_name)"></env>
        </node>

        <node if="$(arg show)" name="image_view" pkg="image_view" type="image_view">
            <remap from="image" to="$(arg show_topic)" />
        </node>

    </group>

</launch>
